cmd/capibmadm/cmd/powervs/port/list.go

```diff
@@ -0,0 +1,137 @@
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package port
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	powerClient "github.com/IBM-Cloud/power-go-client/clients/instance"
+
+	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/iam"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/powervs"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/printer"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+)
+
+var network string
+
+// ListCommand powervs port list command.
+func ListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List PowerVS ports",
+		Example: `
+ # List ports in PowerVS Network
+ export IBMCLOUD_API_KEY=<api-key>
+ capibmadm powervs port list --instance-id <service-instance-id> --network <network ID or name>`,
+	}
+
+	cmd.Flags().StringVar(&network, "network", "", "Network ID or Name(preference will be given to the ID over Name)")
+	_ = cmd.MarkFlagRequired("network")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := listPorts(cmd.Context(), network); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return cmd
+}
+
+func listPorts(ctx context.Context, network string) error {
+
+	log := logf.Log
+	log.Info("Listing PowerVS ports", "service-instance-id", options.GlobalOptions.ServiceInstanceID, "network", network)
+
+	auth := iam.GetIAMAuth()
+	accountID, _ := utils.GetAccountID(ctx, auth)
+	sess, err := powervs.NewPISession(accountID, options.GlobalOptions.PowerVSZone, options.GlobalOptions.Debug)
+	if err != nil {
+		return err
+	}
+
+	c := powerClient.NewIBMPINetworkClient(ctx, sess, options.GlobalOptions.ServiceInstanceID)
+	networks, err := c.GetAll()
+	if err != nil {
+		return fmt.Errorf("failed to get the networks, err: %v", err)
+	}
+
+	var networkNames, networkIDs []string
+	for _, net := range networks.Networks {
+		networkIDs = append(networkIDs, *net.NetworkID)
+		networkNames = append(networkNames, *net.Name)
+	}
+
+	var netID string
+
+	// To get the network ID in case network name is passed
+	if utils.Contains(networkIDs, network) {
+		netID = network
+	} else if utils.Contains(networkNames, network) {
+		for _, n := range networks.Networks {
+			if *n.Name == network {
+				netID = *n.NetworkID
+			}
+		}
+	} else {
+
+		return fmt.Errorf("Not able to find network: \"%s\" by ID or name in the network list", network)
+	}
+
+	portsList, err := c.GetAllPorts(netID)
+	if err != nil {
+		return err
+	}
+
+	listByVersion := PortList{
+		Items: []PortSpec{},
+	}
+
+	for _, port := range portsList.Ports {
+		listByVersion.Items = append(listByVersion.Items, PortSpec{
+			Description: *port.Description,
```

Comment:
Need to nil validate before accessing.
It's better to use the utils.DereferencePointer to nil validate and dereference the pointer.

Reply:
Changed to use the DereferencePointer.